### PR TITLE
New Bot for Fireeye Appliances

### PR DESCRIPTION
**Pull request by CysihZ**
Referring to my mail I created a Parser& Bot which is able to Collect and parse data from fireeye mail and file analysis appliances.
We are collection sha1 and MD5 hashed and if there is network communication we are also collecting urls and domains.

### CODE REVIEW

**Comment by schacht-certat** [comment]
Hi @CysihZ,

please squash your commits and write a meaningful commit message. There are also some failing tests from the spelling checker you might want to look at.

**Comment by wagner-certat** [comment]
And something appears to be wrong with `docs/user/bots.rst `

On the first *glance*, the tests look good :)

File: intelmq/bots/parsers/fireeye/parser.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+Fireeye Parser Bot
+Retrieves a base64 encoded JSON-String from raw and converts it into an
+event.
+"""
+from intelmq.lib import utils
+from intelmq.lib.bot import ParserBot
+from intelmq.lib.message import MessageFactory
+from intelmq.lib.utils import base64_decode
+import json
+import xmltodict
```
Comment (wagner-certat) [inline intelmq/bots/parsers/fireeye/parser.py:12]: In order to make the basic test suite work, wrap this in a try-except please. And the same logic for the init-method as for the collector applies here as well.

File: intelmq/tests/bots/collectors/fireeye/test_collector.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+PARAMETERS = {'dns_name': 'myfireeye.local', 'http_username': RANDSTR, 'http_password': RANDSTR, 'logging_level': 'DEBUG', 'request_duration': '24_hours', 'name': 'FireeyeCollector'}
+
+
+def prepare_mocker(mocker):
+    mocker.post('https://myfireeye.local/wsapis/v2.0.0/auth/login', headers={'X-FeApi-Token': '1234567890'})
+    mocker.get('https://myfireeye.local/wsapis/v2.0.0/alerts?duration=24_hours', text=ASSET_PATH_FIRST.read_text())
+    mocker.get('https://myfireeye.local/wsapis/v2.0.0/openioc?alert_uuid=1591de22-4926-4124-b3ed-ffff96766295', text=ASSET_PATH_SECOND.read_text())
+
+
+@requests_mock.Mocker()
```
Comment (wagner-certat) [inline intelmq/tests/bots/collectors/fireeye/test_collector.py:22]: Please add a `test.skip_exotic()` above this line. The same applies to the parser test. Then, the tests should also be working if the optional library xmltodict is not installed.

**Review by wagner-certat — CHANGES_REQUESTED** [review]
In order to make the testsuite work in all cases, small adaptions are necessary. Will do a full review afterwards.

**Comment by wagner-certat** [comment]
Is the code itself ready? If yes, I can fix the remaining issues myself as they are not bot-related by cherry-picking the relevant commits, eventually squashing them together, fixing merge conflicts.

**Comment by CysihZ** [comment]
Hi, 
sorry for the delay. 
Sounds great, yes  the code itself  is ready!

**Comment by wagner-certat** [comment]
Incorporated in and obsoleted by #1929
Thanks again!

**Pull request closed without merging**